File: typescript/resistor-color-duo/resistor-color-duo.ts
type Color = typeof COLORS[number];

const COLORS = [
  'black', 'brown', 'red', 'orange', 'yellow',
  'green', 'blue', 'violet', 'grey', 'white',
] as const;

export function decodedValue(colors: Color[]): number {
  return COLORS.indexOf(colors[0]) * 10 + COLORS.indexOf(colors[1]);
}

File: python/resistor-color-duo/resistor_color_duo.py
mapping = [
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white",
]


def value(colors):
    return mapping.index(colors[0]) * 10 + mapping.index(colors[1])
